### PR TITLE
[Add]: Relationship between KubeArmorPolicy and Kubernetes Components

### PR DESCRIPTION
**Pull request by pandeyshubham03**
**Notes for Reviewers**

This PR fixes #
1. Adds Inventory Relationship between KubeArmorPolicy and Kubernetes Components

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by pandeyshubham03** [comment]
@MUzairS15 Kindly update the [spreadsheet](https://docs.google.com/spreadsheets/d/1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdwizOJmeMw/edit#gid=1646054916) with the relationships

**Comment by MUzairS15** [comment]
Sure @pandeyshubham03 
With this merged we will be completing the KubeArmor model.

**Comment by pandeyshubham03** [comment]
> Sure @pandeyshubham03 With this merged we will be completing the KubeArmor model.

One more component left i.e., KubeArmorHostPolicy, will be done by today

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
+{
+    "apiVersion": "core.meshery.io/v1alpha1",
+    "kind": "Hierarchical",
+    "metadata": {
+      "description": "A hierarchical inventory relationship in which the configuration of (parent) component is patched with the configuration of other (child) component. Eg: The configuration of the EnvoyFilter (parent) component is patched with the configuration as received from WASMFilter (child) component."
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:5]: Please update the description

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+              "patch": {
+                "patchStrategy": "replace",
+                "mutatedRef": [
+                  [
+                    "annotations",
+                    "_", 
+                    "security.kubearmor.com/policy"
+                  ]
+                ],
+                "description": "Pods in Kubernetes are subjects to KubeArmorPolicy, enforcing security measures on the Pods' system calls and resource access, enhancing the overall security posture of containerized workloads."
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:50]: Update the description to be more oriented towards how Pod behaviour changes when annotated with KubeArmorPolicy .

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "CronJob",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:151]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "Job",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:136]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "ReplicaSet",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:106]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "StatefulSet",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:91]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "Deployment",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:76]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "PodTemplate",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:61]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+          "to": [
+            {
+              "kind": "Pod",
+              "model": "kubernetes",
+              "patch": {
+                "patchStrategy": "replace",
+                "mutatedRef": [
+                  [
+                    "annotations",
+                    "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:46]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+            },
+            {
+                "kind": "Daemonset",
+                "model": "kubernetes",
+                "patch": {
+                  "patchStrategy": "replace",
+                  "mutatedRef": [
+                    [
+                      "annotations",
+                      "_",
```
Comment (MUzairS15) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:121]:
```suggestion
```

File: server/meshmodel/kubearmor/relationships/hierarchical_inventory.json [hunk 11]
```diff
@@ -0,0 +1,158 @@
+{
+    "apiVersion": "core.meshery.io/v1alpha1",
+    "kind": "Hierarchical",
+    "metadata": {
+      "description": "A hierarchical inventory relationship in which the configuration of (parent) component is patched with the configuration of other (child) component. Eg: The configuration of the Pod (parent) component is patched with the configuration as received from KubeArmorPolicy (child) component."
```
Comment (leecalcote) [inline server/meshmodel/kubearmor/relationships/hierarchical_inventory.json:5]:
```suggestion
      "description": "A hierarchical inventory relationship in which the configuration of (parent) component is patched with the configuration of other (child) component. In this case, the configuration of the Kubernetes Pod (parent) component is patched with the configuration as received from KubeArmorPolicy (child) component."
```

**Review by leecalcote — DISMISSED** [review]
I made a slight tweak in the verbiage. Looks good to me.

**Comment by leecalcote** [comment]
@pandeyshubham03 there's a merge conflict to look after...

**Comment by pandeyshubham03** [comment]
> @pandeyshubham03 there's a merge conflict to look after...
The merge conflict is resolved!

**Comment by leecalcote** [comment]
Yay!

**Comment by leecalcote** [comment]
Does the relationships contrib doc capture all your steps?

**Pull request merged**